src/lu.jl

```diff
@@ -102,6 +102,25 @@ function lu!(A::HermOrSym{T}, pivot::Union{RowMaximum,NoPivot,RowNonZero} = lupi
     end
     lu!(A.data, pivot; check, allowsingular)
 end
+
+#reusing LU object
+#lu!(F::LU,A) should be dispatched on the type of matrix stored in the LU factorization.
+
+function lu!(F::LU{<:Any,<:StridedMatrix{<:BlasFloat}}, A; check::Bool = true, allowsingular::Bool = false)
+    copyto!(F.factors,A)
+    lpt = LAPACK.getrf!(F.factors, F.ipiv; check)
+    check && _check_lu_success(lpt[3], allowsingular)
+    return F
```

Comment:
Shouldn't this be something like
```suggestion
    return LU{T,typeof(lpt[1]),typeof(lpt[2])}(lpt[1], lpt[2], lpt[3])
```
as otherwise it doesn't update the `info` field?

Returning a new LU object should be cheap since it is immutable and just contains pointers to the same arrays.